test/Microsoft.AspNetCore.SignalR.Tests/HubEndpointTests.cs

```diff
@@ -370,6 +370,28 @@ public class HubEndpointTests
             }
         }
 
+        [Fact]
+        public async Task HandshakeFailureFromIncompatibleProtocolVersionSendsResponseWithError()
+        {
+            var hubProtocolMock = new Mock<IHubProtocol>();
+            hubProtocolMock.Setup(m => m.Name).Returns("json");
+            hubProtocolMock.Setup(m => m.Version).Returns(9001);
```

Comment:
![https://media1.tenor.com/images/5d3022e7d8f6b0e36b86170b676c601c/tenor.gif?itemid=4474083](https://media1.tenor.com/images/5d3022e7d8f6b0e36b86170b676c601c/tenor.gif?itemid=4474083)

Reply:
That GIF is insufficiently exclamation-marked.